### PR TITLE
feat(react): export react component types

### LINKED ISSUE #307: Export component types
**Is your feature request related to a problem? Please describe.**
Sometimes when creating a custom component using existing Baklava component you could spread props. It would be great to import types of e.g "BlInputType" so we could use it as a type and spread. I'm open for suggestions if any :)

**Describe the solution you'd like**
Export `BlInputType` and other component types from `baklava-react.d.ts`

### PR DESCRIPTION
**Pull request by AykutSarac**
This PR adds support to use React components both as value and type.

Example usage:
```tsx
import { BlDialog } from "@trendyol/baklava/dist/baklava-react";

const Component = () => {
  const dialogRef = React.useRef<BlDialog>(null);

  return (
    <BlDialog ref={dialogRef}>...</BlDialog>
  )
}
```

Fixes #307

### CODE REVIEW

**Comment by muratcorlu** [comment]
Do you see any risk of this change to break current React implementations?

**Comment by AykutSarac** [comment]
> Do you see any risk of this change to break current React implementations?

No, it only brings type support which was previously inferred as `any`. There shouldn't be any breaking change.

**Review by erbilnas — APPROVED** [review]

**Review by muratcorlu — APPROVED** [review]

File: scripts/generate-react-exports.js [hunk 1]
```diff
@@ -77,6 +80,7 @@ for (const module of customElementsModules) {
   const Type = componentName + 'Type';
 
   importStatements.push(`import type ${Type} from "./${importPath}";`);
+  exportStatements.push(`export declare type ${Type.replace('Type', '')} = ${Type}`);
```
Comment (Enes5519) [inline scripts/generate-react-exports.js:83]: You can use `componentName` variable instead of `${Type.replace('Type', '')}`.

Comment (AykutSarac) [reply scripts/generate-react-exports.js]: Nice catch, updated.

**Pull request merged**